regexec.c

```diff
@@ -229,7 +228,7 @@ S_regcppush(pTHX_ const regexp *rex, I32 parenfloor, U32 maxopenparen _pDEPTH)
 {
     const int retval = PL_savestack_ix;
     const int paren_elems_to_push =
-                (maxopenparen - parenfloor) * REGCP_PAREN_ELEMS;
+        (sizeof(*rex->offs) * (maxopenparen - parenfloor)) / sizeof(*PL_savestack);
     const UV total_elems = paren_elems_to_push + REGCP_OTHER_ELEMS;
```

Comment:
You are mixing units in the calculation of `total_elems`, so the `SSGROW` call below will use an incorrect argument. As a result, the savestack will not be grown as much as is needed.

Reply:
I don't quite get that? - I think I solved potential alignment issues with my second patch.

Reply:
It is not an alignment issue, but a size issue (or units issue).

Previously, `paren_elems_to_push` and `total_elems` were both measured in units of `sizeof(IV)`. With this change, `paren_elems_to_push` is now measured in units of `sizeof(ANY)`. So `total_elems` is now a nonsense value calculated by adding two things measured in different units.

Reply:
Aren't they the same thing?

Reply:
I guess not but I measure the elements needed in the stack nothing more -  taking the stack element size.

Reply:
I guess if you mean in terms of code clarity and the later if for debug printing then I may just adda another variable.

Reply:
Looking again, I may be totally wrong (I think I misinterpreted the `SSGROW`), and that `paren_elems_to_push` and `total_elems` were always actually measured in units of `sizeof(ANY)`.

I'm looking further.

Reply:
Ok, I was wrong, my apologies for the confusion.

It would probably be useful to add a comment something like:
```
    /* Number of savestack[] entries to be filled by the paren data */
    const int paren_elems_to_push = ...
```

Reply:
Added and reworked debugging prints.